opencti-platform/opencti-graphql/src/database/middleware.js

```diff
@@ -1989,7 +1989,11 @@ const createRuleDataPatch = (instance) => {
   if (isBasicRelationship(instance.entity_type)) {
     patch.i_inference_weight = weight;
   }
-  const attributes = RULES_ATTRIBUTES_BEHAVIOR.supportedAttributes();
+  let attributes = RULES_ATTRIBUTES_BEHAVIOR.supportedAttributes();
+  if (instance.entity_type) { // keep only supported attributes
+    const entityAttributes = schemaAttributesDefinition.getAttributeNames(instance.entity_type);
+    attributes = attributes.filter((attr) => entityAttributes.includes(attr));
```

Comment:
I think entityAttributes doesnt contains refs, so objectMarkings will be excluded and your tests will fail :)

Reply:
I changed  RULES_ATTRIBUTES_BEHAVIOR to get supported attributes by entity type based on all existing rules and which entities they create.